docs/2.7.0/docs/app-dev/explicit-contract-disclosure.rst

```diff
@@ -0,0 +1,277 @@
+be constructed from the fields of the same name from the original contract's ``CreatedEvent``.
+
+.. note:: Only contracts created starting with Canton 2.6 can be shared as disclosed contracts.
+  Prior to this version, contracts' **CreatedEvent** does not have ``ContractMetadata`` populated
+  and cannot be used as disclosed contracts.
+
+.. _submitter-disclosed-contract:
+
+Attaching a disclosed contract to a command submission
+------------------------------------------------------
+
+A disclosed contract can be attached as part of the ``Command``'s :ref:`disclosed_contracts <com.daml.ledger.api.v1.Commands.disclosed_contracts>`
+and requires the following fields (see :ref:`DisclosedContract <com.daml.ledger.api.v1.DisclosedContract>` for content details) to be populated from
+the original `CreatedEvent` (see above):
+
+- **template_id** - The contract's template id.
+- **contract_id** - The contract id.
+- **arguments** - The contract's create arguments. This field is a protobuf ``oneof``
+  and it allows either passing the contract's create arguments typed (as ``create_arguments``)
+  or as a byte array (as ``create_arguments_blob``).
+  Generally, clients should use the ``create_arguments_blob`` for convenience since they can be received as such
+  from the stakeholder off-ledger (see above).
+- **metadata** - The contract metadata. This field can be populated as received from the stakeholder (see below).
+
+Trading the stock with explicit disclosure
+-------------------------------------------------
+
+In the example above, **Buyer** does not have visibility over the ``stockCid``, ``priceQuotationCid`` and ``offerCid`` contracts,
+so **Buyer** must provide them as disclosed contracts in the command submission exercising ``Offer_Accept``. To
+do so, the contracts' stakeholders must fetch them from the ledger and make them available to the **Buyer**.
+
+.. note:: Daml Script support for explicit disclosure is currently not implemented.
+  The last steps of the example are modeled using raw gRPC queries.
+
+The contracts' stakeholders issue fetch queries to the Ledger API for retrieving
+the associated contract payloads. For simplicity in the example, all parties reside on participant ``participant``
+with the Ledger API running on port ``5031``.
+
+::
+
+  -- StockExchange fetches the Stock contract referenced by stockCid from the ledger by querying the Ledger API
+  -- (here we are using the GetTransactions query)
+  grpcurl -plaintext -d '{"ledgerId":"participant","begin":{"absolute":"0000000000000000"},"end":{"boundary":"LEDGER_END"},"filter":{"filtersByParty":{"StockExchange::12206edc46b86d9ea5f75407fa04720ea005f253ca736fb047a8c25839edf2fbb8fa":{"inclusive":{"templateIds":[{"packageId":"5811603b1342a2a791fd69aff08d710f67bf652f3e7129dd51f69bc01bb30f19","moduleName":"StockExchange","entityName":"Stock"}]}}}},"verbose":true}' localhost:5031 com.daml.ledger.api.v1.TransactionService/GetTransactions
```

Comment:
UUIDs don't make for nice reading. One alternative would be to redirect output to `.json` files then re-combine them with `jq` before piping into the next `grpcurl` command. The downside of this approach is that most readers will not be familiar with advanced `jq` hacking.